python-checks/src/main/java/org/sonar/python/checks/IgnoredPureOperationsCheck.java

```diff
@@ -298,10 +298,12 @@ private static void checkExpression(SubscriptionContext ctx, Expression expressi
     if (expression.is(Tree.Kind.CALL_EXPR)) {
       CallExpression callExpression = (CallExpression) expression;
       PythonType pythonType = callExpression.callee().typeV2();
-      PURE_FUNCTIONS.stream()
-        .filter(f -> typeChecker.typeCheckBuilder().isTypeWithName(f).check(pythonType).equals(TriBool.TRUE))
-        .findFirst()
-        .ifPresent(result -> ctx.addIssue(callExpression.callee(), String.format(MESSAGE_FORMAT, result)));
+      if (typeChecker.typeCheckBuilder().isObjectType().check(pythonType) == TriBool.FALSE) {
```

Comment:
I think extracting this condition into its own method would help readability as it can be named

Reply:
(I removed the condition altogether)